mii/server.py

```diff
@@ -102,242 +80,81 @@ def _is_server_process_alive(self, process):
-                          deployment_name,
-                          model_name,
-                          model_path,
-                          ds_optimize,
-                          ds_zero,
-                          ds_config,
-                          mii_configs,
-                          hostfile,
-                          host,
-                          port,
-                          master_port,
-                          deploy_ranks):
+    def _generate_ds_launch_str(self, replica_config, hostfile):
         # use different hostfiles for replica instances
         # pass /dev/null when no replica is used
         worker_str = f"-H {hostfile} "
         # pin deepspeed launch to specific gpu id(s)
-        included_gpus = f"{host}:{','.join(map(str, deploy_ranks))}"
+        included_gpus = f"{replica_config.hostname}:{','.join(map(str, replica_config.gpu_indices))}"
         worker_str += f"-i {included_gpus} "
 
         # adjust torch dist port depending on rank, otherwise multi-replica deployments will conflict
         # assign different ports to replicas because they could be on the same host
-        worker_str += f"--master_port {master_port}"
+        worker_str += f"--master_port {replica_config.torch_dist_port}"
 
         ds_launch_str = f"deepspeed {worker_str} --no_local_rank --no_python"
 
-        return self._launch_server_process(deployment_name,
-                                           model_name,
-                                           model_path,
-                                           ds_optimize,
-                                           ds_zero,
-                                           ds_config,
-                                           mii_configs,
-                                           port,
-                                           "MII server",
-                                           ds_launch_str=ds_launch_str)
-
-    def _initialize_service(self,
-                            deployment_name,
-                            model_name,
-                            model_path,
-                            ds_optimize,
-                            ds_zero,
-                            ds_config,
-                            mii_configs,
-                            lb_config):
+        return ds_launch_str
 
+    def _initialize_service(self, mii_config):
         processes = []
+        server_args = [
+            f"--load-balancer-port {mii_config.port_number}",
+            f"--restful-gateway-port {mii_config.restful_api_port}",
+        ]
 
         host_gpus = defaultdict(list)
-        for repl_config in lb_config.replica_configs:
+        for repl_config in mii_config.deployment_config.replica_configs:
             host_gpus[repl_config.hostname].extend(repl_config.gpu_indices)
 
         # Start replica instances
-        for i, repl_config in enumerate(lb_config.replica_configs):
+        for repl_config in mii_config.deployment_config.replica_configs:
```

Comment:
I refactored this to remove the `_launch_{X|Y|Z}()` methods - but this this function got a bit long in the process. It may be cleaner to move back to the individual functions.